cgen: Incorrectly generated code when `in array init` at the if cond

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.3 3345a17
OS: windows,  v 
Processor: 16 cpus, 64bit, little endian, Error: 'wmic' is not recognized as an internal or external command,

operable program or batch file.
```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg main.v`
See the attached file: [main.v.zip](https://github.com/vlang/v/files/13790203/main.v.zip)


Here's the code that I assume is leading to the error:
```v
for seed in seeds {
	mut soil := seed
	for key in data['seed-to-soil'].keys() {
		if seed in []u32{len: int(data['seed-to-soil'][key]['range']), init: u32(index) + key} {
			soil = data['seed-to-soil'][key]['destination'] + seed - key
			break
		}
	}
	...
}
```

I was experimenting with this bit:
```v
if seed in []u32{len: int(data['seed-to-soil'][key]['range']), init: u32(index) + key}
```

**What did you expect to see?**

Successful compilation.


**What did you see instead?**

From inspecting the referenced `main.15584995340835800118.tmp.c:18561: error: '(' expected (got "Array_u32_contains")`, it seems that no parenthesis are being added to the if statement:

```c
if Array_u32_contains(_t4, seed) {
    soil = (u32)((u32)((*(u32*)map_get(ADDR(map, (*(Map_string_u32*)map_get(ADDR(map, (*(Map_u32_Map_string_u32*)map_get(ADDR(map, data), &(string[]){_SLIT("seed-to-soil")}, &(Map_u32_Map_string_u32[]){ new_map_noscan_key(sizeof(u32), sizeof(Map_string_u32), &map_hash_int_4, &map_eq_int_4, &map_clone_int_4, &map_free_nop) }))), &(u32[]){key}, &(Map_string_u32[]){ new_map_noscan_value(sizeof(string), sizeof(u32), &map_hash_string, &map_eq_string, &map_clone_string, &map_free_string) }))), &(string[]){_SLIT("destination")}, &(u32[]){ 0 })) + seed) - key);
    break;
}
```

Error message:

```
==================
C:/Users/username/AppData/Local/Temp/v_0/main.15584995340835800118.tmp.c:17611: warning: implicit declaration of function 'CreateSymbolicLinkW'
C:/Users/username/AppData/Local/Temp/v_0/main.15584995340835800118.tmp.c:18561: error: '(' expected (got "Array_u32_contains")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

After simplifying:

```v
fn main() {
	if 0 in []int{} {
	}
}
```

error info:
```
==================
/tmp/v_501/a.7674770451773310630.tmp.c:12660:5: error: expected '(' after 'if'
        if Array_int_contains(__new_array_with_default_noscan(0, 0, sizeof(int), 0), a) {
           ^
1 error generated.
...
==================
(Use `v -cg` to print the entire error message)
```

--- a/vlib/v/gen/c/infix.v
+++ b/vlib/v/gen/c/infix.v
@@ -518,7 +518,9 @@ fn (mut g Gen) infix_expr_in_op(node ast.InfixExpr) {
 						expr: node.left
 						expr_type: ast.mktyp(node.left_type)
 					}
+					g.write('(')
 					g.gen_array_contains(node.right_type, node.right, elem_type, new_node_left)
+					g.write(')')
 					return
 				}
 			} else if elem_type_.sym.kind == .interface_ {
@@ -528,11 +530,15 @@ fn (mut g Gen) infix_expr_in_op(node ast.InfixExpr) {
 					expr: node.left
 					expr_type: ast.mktyp(node.left_type)
 				}
+				g.write('(')
 				g.gen_array_contains(node.right_type, node.right, elem_type, new_node_left)
+				g.write(')')
 				return
 			}
 		}
+		g.write('(')
 		g.gen_array_contains(node.right_type, node.right, node.left_type, node.left)
+		g.write(')')
 	} else if right.unaliased_sym.kind == .map {
 		g.write('_IN_MAP(')
 		if !left.typ.is_ptr() {
@@ -597,17 +603,23 @@ fn (mut g Gen) infix_expr_in_op(node ast.InfixExpr) {
 						expr: node.left
 						expr_type: ast.mktyp(node.left_type)
 					}
+					g.write('(')
 					g.gen_array_contains(node.right_type, node.right, elem_type, new_node_left)
+					g.write(')')
 					return
 				}
 			}
 		}
+		g.write('(')
 		g.gen_array_contains(node.right_type, node.right, node.left_type, node.left)
+		g.write(')')
 	} else if right.unaliased_sym.kind == .string {
+		g.write('(')
 		g.write('string_contains(')
 		g.expr(node.right)
 		g.write(', ')
 		g.expr(node.left)
+		g.write(')')
 		g.write(')')
 	}
 }

--- a/vlib/v/tests/if_expression_test.v
+++ b/vlib/v/tests/if_expression_test.v
@@ -263,3 +263,13 @@ fn test_noreturn() {
 		assert_false_noreturn()
 	}
 }
+
+// for issue 20300
+// Phenomenon of issue:
+// infix expr generates wraparound parentheses, but misses the case where `array_contains()` is used.
+fn test_in_array_init() {
+	if 0 in []int{} {
+		assert false
+	}
+	assert true
+}

